### ISSUE TITLE
Keyboard Navigation

### ISSUE THREAD

**Issue by diego-carrera** [sheodox/alexandrite#17]
It would be great to have some sort of keyboard navigation.

- Navigate to Previous and Next Post
- Expand images or content
- Open comments
- Go to the linked content

**Comment by adrianzech** [comment]
Navigate to Previous and Next Post would be awesome!

Maybe also a shorcut to open the sidebar/navigation bar?

**Comment by dtantono** [comment]
I hope we can hold ctrl with left click to open the new thread in new tab like default lemmy.

**Comment by sheodox** [comment]
@dtantono Good catch, I'll add some code to make sure I'm not interfering with the ctrl+click functionality. In the meantime you can click down the scroll wheel on your mouse or right click and hit 'Open in new tab' to do that.

**Comment by barelylit** [comment]
+1 for me on this

Keyboard navigation is one of my most missed features of RES since moving to Lemmy, this would be amazing.

**Comment by sheodox** [comment]
What sort of hotkeys were you all thinking? Like which keys should do what. I surprisingly never really looked into the hotkeys for reddit/RES.

**Comment by barelylit** [comment]
Next post / previous post
expand / close
next / last image in a gallery
go to comments were my main ones. 

Here's all the ones in RES

![image](https://github.com/sheodox/alexandrite/assets/101836977/cdd27a5a-ff84-4d77-8296-62e4641c7962)
![image](https://github.com/sheodox/alexandrite/assets/101836977/09211ffe-f96e-469b-a165-6bf23c03d04d)

**Comment by sheodox** [comment]
I just added a handful of hotkeys. You can see what ones I have on the new [hotkey help](https://alexandrite.app/help/hotkeys) page. I'll likely add some more over time, but some basic ones are in there now!

**Comment by barelylit** [comment]
Dont seem to be working here on Brave or Chrome,

J and K intermittently make things jump about, but no coherent movement between posts

**Comment by sheodox** [comment]
@barelylit I just tried Chrome and it's working fine for me. Could you share a video of what you're seeing?

**Comment by sheodox** [comment]
Also note, that when a post is open and focus is somewhere in the post (column or overlay) `j` and `k` will jump between comments instead.

**Comment by barelylit** [comment]
So it seems to be that if there is a sticky post, that is locking it, till you keyboard down arrow past it then hot keys work as expected 

I'm hitting J here constantly, at the start but it has no effect till ive scrolled past the sticky posts.

https://github.com/sheodox/alexandrite/assets/101836977/2b14c18d-4003-404c-8efe-28601cabf701

**Comment by barelylit** [comment]
Actually, doesnt have to be a sticky post, just have to be scrolled down past the first post before it activates

**Comment by Kommynct** [comment]
What is really needed is just settings that let you change the keyboard navigation keys. i'm on dvorak and a lot of it is nonsense on this layout.

**Comment by InkstainTheBat** [comment]
Agreed. Any updates?
Edit: I'd also love to see greater functionality for navigating on keyboard overall: things like a reply hotkey, a collapse hotkey, better selected item highlighting, bottom comment navigation (I can't navigate to and interact with the bottom-most comments), or not having to click on an empty section after clicking somewhere else to be able to use hotkeys again (aka making them work even when comments aren't focused)
